lib/model/game_state.dart

```diff
@@ -70,6 +66,10 @@ class GameState with ChangeNotifier {
       deck.add(card);
     }
 
+    if (scrying.isEmpty) {
+      nextCard();
+    }
```

Comment:
Now the scry action suppresses this, so we do it here at the point where all scrying is done.